core/src/Cuda/Kokkos_Cuda_Half.hpp

```diff
@@ -47,12 +47,13 @@
 
 #include <Kokkos_Macros.hpp>
 #ifdef KOKKOS_ENABLE_CUDA
+#if !(defined(KOKKOS_COMPILER_CLANG) && KOKKOS_COMPILER_CLANG < 900) && \
+    !defined(KOKKOS_ARCH_KEPLER35)
```

Comment:
we probably need others here too. My guess is everything below Pascal. That means Kepler37, Maxwell50 and Maxwell52

Reply:
Just try the compilation with all the archs and see which one works

Reply:
Maxwell52, Maxwell50, Keppler37, Keppler32, Keppler30 fail. Issuing PR.